sql/parse/parse.go

```diff
@@ -938,22 +944,51 @@ func isAggregateFunc(v *sqlparser.FuncExpr) bool {
 	return v.IsAggregate()
 }
 
+// Convert an integer, represented by the specified string in the specified
+// base, to its smallest representation possible, out of:
+// int8, uint8, int16, uint16, int32, uint32, int64 and uint64
+func convertInt(value string, base int) (sql.Expression, error) {
+	i8, err := strconv.ParseInt(value, base, 8)
```

Comment:
Instead of the great pyramid of Ghiza what do you think of doing this to keep it a little bit more flat? 😛 

```go
if i8, err := stconv.ParseInt(value, base, 8); err == nil {
    return expression.NewLiteral(int8(i8), sql.Int8), nil
}

if ui8, err := stconv.ParseUint(value, base, 8); err == nil {
    return expression.NewLiteral(unt8(ui8), sql.Uint8), nil
}

// the rest of the checks

ui64, err := strconv.ParseUint(value, base, 64)
if err != nil {
    return nil, err
}

return expression.NewLiteral(uint64(ui64), sql.Uint64), nil
```

Reply:
This is what I wanted to hear, that arrow shape was ugly. Thanks!

Reply:
Done in https://github.com/src-d/go-mysql-server/pull/840/commits/a7aa47f53e96d9088894e3f9efe03f6e2ac4ef83